test/Tests.Infrastructure/TestBase.cs

```diff
@@ -127,9 +130,20 @@ protected RavenServer GetNewServer(IDictionary<string, string> customSettings =
 
                 configuration.Initialize();
                 configuration.DebugLog.LogMode = LogMode.None;
-                if (customSettings == null || customSettings.ContainsKey("Raven/ServerUrl") == false)
+
+                if (_ports.Count == 0)
+                {
+                    for (int i = 0; i < 1000; i++)
+                    {
+                        _ports.Enqueue(8090 + i);
+                    }
+                }
+
+                if (customSettings?.ContainsKey("Raven/ServerUrl") == false)
                 {
-                    configuration.Core.ServerUrl = "http://127.0.0.1:0";
+                    int port;
+                    _ports.TryDequeue(out port);
+                    configuration.Core.ServerUrl = "http://127.0.0.1:" + port;
```

Comment:
No, that is a very bad idea. This will force us to use the same port, instead of letting the OS pick them for us.